labs/src/main/scala/org/scalalabs/basic/lab01/OOExercise.scala

```diff
@@ -40,6 +40,31 @@ import scala.language.implicitConversions
  *   of type [[org.scalalabs.basic.lab01.CurrencyConverter]]
  * - Use the implicit CurrencyConverter to do the conversion. 
  */
-class Euro {
+abstract class Currency(val symbol:String)
 
+class Dollar(val dollar: Int, val cents: Int = 0) extends Currency("USD") {
+  val inCents = (dollar * 100) + cents
+}
+
+class Euro(val euro: Int, val cents: Int = 0) extends Currency("EUR") with Ordered[Euro] {
+  
+  val inCents = (euro * 100) + cents
+  
+  def +(that: Euro): Euro = Euro fromCents (this.inCents + that.inCents)
+  def *(scalar: Int): Euro = Euro fromCents (scalar * inCents)
+  
+  def compare(that: Euro) = this.inCents - that.inCents
+  
+  override def toString = s"$symbol: $euro,${if (cents == 0) "--" else f"$cents%02d"}"
```

Comment:
Interesting; I didn’t realize you could do that complex of an expression in string interpolation. Especially with the nesting.